backend/metadata_writer/src/metadata_writer.py

```diff
@@ -80,7 +80,8 @@ def patch_pod_metadata(
 
 def output_name_to_argo(name: str) -> str:
     import re
-    return re.sub('-+', '-', re.sub('[^-0-9a-z]+', '-', name.lower())).strip('-')
+    # This sanitization code should be kept in sync with the code in the DSL compiler.
```

Comment:
Can you put a link to related DSL compiler code?

Reply:
Do you want me to put it in the comment? 
https://github.com/kubeflow/pipelines/blob/39975e3cde7ba4dcea2bca835b92d0fe40b1ae3c/sdk/python/kfp/compiler/_k8s_helper.py#L33

Reply:
Done.